drivers/spi/spi_mcux_lpspi.c

```diff
@@ -276,7 +283,52 @@ static void spi_mcux_dma_callback(const struct device *dev, void *arg,
 			data->status_flags |= SPI_MCUX_LPSPI_DMA_ERROR_FLAG;
 		}
 	}
+#if CONFIG_SPI_ASYNC
+	if (data->ctx.asynchronous) {
+		const struct spi_mcux_config *config = spi_dev->config;
+
+		if ((data->status_flags & SPI_MCUX_LPSPI_DMA_DONE_FLAG)  ==
+			SPI_MCUX_LPSPI_DMA_DONE_FLAG) {
+			while ((LPSPI_GetStatusFlags(config->base) & kLPSPI_ModuleBusyFlag)) {
+			/* wait until module is idle */
```

Comment:
No longer blocking, originally wanted to block here because I was unsure how the spi and dma would react to high traffic, tested with custom sample, should be good.

Reply:
I am concerned with how easily we are making changes to code that has no explanation